yupiik-batch-runtime/src/test/java/io/yupiik/batch/runtime/batch/builder/BatchChainTest.java

```diff
@@ -44,4 +46,25 @@ public String name() {
         };
         assertInstanceOf(TimeoutException.class, assertThrows(IllegalStateException.class, () -> infiniteStep.run(configuration)).getCause());
     }
+
+    @Test
+    void failOnPromiseError() {
```

Comment:
likely need another test ensuring it does not fail the other way around, the easier being to  use a two steps chain where first one will be returned completed (as you did here) so will be immediately removed (the falg being false) so the batch will NOT fail.
Then this test must set the toggle to true and ensure we fail in await probably.

Reply:
adding success step in tests